### PR TITLE
ESC9A Post Processing

### PR DESCRIPTION
**Pull request by rvazarkar**
## Description

This MR implements post processing for ADCS Escalation 9a

## Motivation and Context

See description

## How Has This Been Tested?
Tested using lab environment as well as with integration tests.

## Screenshots (if appropriate):
image

## Types of changes

-   [ ] Chore (a change that does not modify the application functionality)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Documentation updates are needed, and have been made accordingly.
-   [x] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: packages/go/analysis/ad/adcs.go [hunk 1]
```diff
@@ -309,6 +309,16 @@ func PostADCS(ctx context.Context, db graph.Database, groupExpansions impact.Pat
 
 						return nil
 					})
+
+					operation.Operation.SubmitReader(func(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob) error {
+						if err := PostADCSESC9a(ctx, tx, outC, groupExpansions, innerEnterpriseCA, innerDomain, cache); err != nil {
+							log.Errorf("failed post processing for %s: %v", ad.ADCSESC3.String(), err)
```
Comment (JonasBK) [inline packages/go/analysis/ad/adcs.go:315]: Replace  `ad.ADCSESC3.String()` with `ad.ADCSESC9a.String()`

File: packages/go/analysis/ad/esc9.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+		return false, nil
+	} else if enrolleeSuppliesSubject, err := ct.Properties.Get(ad.EnrolleeSuppliesSubject.String()).Bool(); err != nil {
+		return false, err
+	} else if enrolleeSuppliesSubject {
+		return false, nil
+	} else if schemaVersion, err := ct.Properties.Get(ad.SchemaVersion.String()).Float64(); err != nil {
+		return false, err
+	} else if authorizedSignatures, err := ct.Properties.Get(ad.AuthorizedSignatures.String()).Float64(); err != nil {
+		return false, err
+	} else if schemaVersion > 1 && authorizedSignatures > 0 {
```
Comment (JonasBK) [inline packages/go/analysis/ad/esc9.go:157]: From the ticket: `SchemaVersion = 1 OR (SchemaVersion > 1 AND AuthorizedSignatures = 0)`
So I guess we should change the check to: `schemaVersion == 1 || (schemaVersion > 1 && authorizedSignatures > 0)`

**Review by JonasBK — CHANGES_REQUESTED** [review]
Made two small comments to the logic but else it looks great!

File: packages/go/analysis/ad/esc9.go (hunk 2 above)
Comment (rvazarkar) [reply packages/go/analysis/ad/esc9.go:157]: This check works as expected. If the schemaversion is 1, it wont drop into that else statement. If the schemaversion is > 1 and authorizedsignatures is > 0 it'll drop into the else statement and return false

Comment (JonasBK) [reply packages/go/analysis/ad/esc9.go:157]: right. I read that statement incorrectly

**Review by JonasBK — APPROVED** [review]

**Pull request merged**